pyiceberg/io/pyarrow.py

```diff
@@ -1068,20 +1068,13 @@ def primitive(self, primitive: pa.DataType) -> PrimitiveType:
             return StringType()
         elif pa.types.is_date32(primitive):
             return DateType()
-        elif isinstance(primitive, pa.Time64Type) and primitive.unit == "us":
+        elif isinstance(primitive, pa.Time64Type) and primitive.unit in ["us", "ns"]:
             return TimeType()
         elif pa.types.is_timestamp(primitive):
             primitive = cast(pa.TimestampType, primitive)
-            if primitive.unit in ("s", "ms", "us"):
+            if primitive.unit in ("s", "ms", "us", "ns"):
                 # Supported types, will be upcast automatically to 'us'
                 pass
-            elif primitive.unit == "ns":
-                if self._downcast_ns_timestamp_to_us:
-                    logger.warning("Iceberg does not yet support 'ns' timestamp precision. Downcasting to 'us'.")
-                else:
-                    raise TypeError(
-                        "Iceberg does not yet support 'ns' timestamp precision. Use 'downcast-ns-timestamp-to-us-on-write' configuration property to automatically downcast 'ns' to 'us' on write."
-                    )
```

Comment:
this is still necessary because iceberg doesnt support `ns`. this workaround help us downcast a column with `ns` to `us`

Reply:
I will add back this code snippet.